kong/plugins/ai-prompt-guard/handler.lua

```diff
@@ -0,0 +1,110 @@
+local _M = {}
+
+-- imports
+local kong_meta = require "kong.meta"
+local buffer    = require("string.buffer")
+--
+
+_M.PRIORITY = 771
+_M.VERSION = kong_meta.version
+
+local function bad_request(msg, reveal_msg_to_client)
+  -- don't let users know 'ai-prompt-guard' is in use
+  kong.log.warn(msg)
+  if not reveal_msg_to_client then
+    msg = "bad request"
+  end
+  return kong.response.exit(400, { error = { message = msg } })
+end
+
+function _M.execute(request, conf)
+  local user_prompt
+
+  -- concat all 'user' prompts into one string, if allowed
+  if request.messages and not conf.allow_all_conversation_history then
+    local buf = buffer.new()
+
+    for k, v in ipairs(request.messages) do
+      if v.role == "user" then
+        buf:put(v.content)
+      end
+    end
+
+    user_prompt = buf:get()
+  elseif request.messages then
+    -- just take the trailing 'user' prompt
+    for k, v in ipairs(request.messages) do
```

Comment:
Unused loop vars should be designated as `_`

```suggestion
    for _, v in ipairs(request.messages) do
```